pkg/controller/resourcequota/resource_quota_controller_patch.go

```diff
@@ -17,70 +17,84 @@ limitations under the License.
 package resourcequota
 
 import (
+	"context"
 	"fmt"
 	"reflect"
-	"time"
 
+	"github.com/kcp-dev/logicalcluster/v2"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
-	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 )
 
-func (rq *Controller) KCPUpdateMonitors(discoveryFunc NamespacedResourcesFunc, period time.Duration, stopCh <-chan struct{}) {
-	// Something has changed, so track the new state and perform a sync.
+func (rq *Controller) KCPUpdateMonitors(ctx context.Context, clusterName logicalcluster.Name, discoveryFunc NamespacedResourcesFunc) {
 	oldResources := make(map[schema.GroupVersionResource]struct{})
-	wait.Until(func() {
```

Comment:
am confused. This basically just removed the loop? To make it much easier to read, replace `wait.Until(func() {` with `func() {` and `}()` at the end, without changing indention. Also has higher chance to be conflict free.

Reply:
Good idea, will do

Reply:
much better